File: src/edu/berkeley/sixtus/simul/Cell.java
package edu.berkeley.sixtus.simul;
/**
 * 
 * @author xvilajosana
 *
 */
public class Cell implements Comparable<Cell> {

	public enum SlotType {
		TX, RX, TXRX, OFF
	}
	
	private int slotNumber;
	private int channelOffset;
	private int target;
	private SlotType type;
	

	public Cell(int slotNumber, int channelOffset, int target, SlotType type) {
		super();
		this.slotNumber = slotNumber;
		this.channelOffset = channelOffset;
		this.target = target;
		this.type = type;
	}
	
	public int getSlotNumber() {
		return slotNumber;
	}
	public void setSlotNumber(int slotNumber) {
		this.slotNumber = slotNumber;
	}
	public int getChannelOffset() {
		return channelOffset;
	}
	public void setChannelOffset(int channelOffset) {
		this.channelOffset = channelOffset;
	}
	public int getTarget() {
		return target;
	}
	public void setTarget(int target) {
		this.target = target;
	}
	public SlotType getType() {
		return type;
	}
	public void setType(SlotType type) {
		this.type = type;
	}

	public String toString(){
		return "CELL:[slot: " + slotNumber + " ch: " + channelOffset + " target: " + target + " type: " +type+"]";
	}

	@Override
	public int compareTo(Cell o) {
		if (this.slotNumber==o.slotNumber &&
				this.channelOffset==o.channelOffset&&
				this.type==o.type&&
				this.target==o.target){
			return 0;
		}
		
		return 1;
		
	}
	
}

File: python/ScheduleSimulator/simulator/Cell.py
'''
Created on 27/06/2013

@author: xvilajosana
'''




class Cell(object):
    '''
    classdocs
    '''
    def __init__(self,timeslot,choff,Ctype,target):
        '''
        Constructor
        '''
        self.slotNumber=timeslot
        self.channelOffset=choff
        self.type=Ctype
        self.target=target
        
        
    def getSlotNumber(self):
        return self.slotNumber

    def setSlotNumber(self, slotNumber):
        self.slotNumber = slotNumber
    
    def getChannelOffset(self):
        return self.channelOffset

    def setChannelOffset(self,channelOffset):
        self.channelOffset = channelOffset

    def getTarget(self):
        return self.target

    def setTarget(self, target) :
        self.target = target

    def getType(self):
        return self.type

    def setType(self, Ctype) :
        self.type = Ctype

    def compareTo(self, o):
        if (self.slotNumber==o.slotNumber and self.channelOffset==o.channelOffset and self.type==o.type and self.target==o.target):
            return 0
        return 1



